Framework/API/src/ExperimentInfo.cpp

```diff
@@ -191,135 +174,13 @@ const std::string ExperimentInfo::toString() const {
 // Helpers for setInstrument and getInstrument
 namespace {
 void checkDetectorInfoSize(const Instrument &instr,
```

Comment:
I opened an issue for this: https://github.com/mantidproject/mantid/issues/20146 (beyond the scope of this PR).